semgrep-core/src/engine/Match_rules.ml

```diff
@@ -430,15 +430,15 @@ let (matches_of_matcher :
               |> List.map (fun (xpat, id, pstr) ->
                      let xs = matcher.matcher target_content file xpat in
                      xs
-                     |> List.map (fun (loc, env) ->
+                     |> List.map (fun (loc1, loc2, env) ->
                             (* this will be adjusted later *)
                             let rule_id = fake_rule_id (id, pstr) in
                             {
                               PM.rule_id;
                               file;
-                              range_loc = (loc, loc);
+                              range_loc = (loc1, loc2);
```

Comment:
From the changes to the Comby and Regex matchers it seems that we could still get a single `loc` from `matcher` here, and instead `loc2` could be computed from that `loc`. But I don't know what `pos2` is in `let (pos1, _), (_, pos2) = m.Spacegrep.Match.region in` in the Spacegrep matcher. If `pos2` cannot be computed from `pos1` and `m.Spacegrep.Match.capture.value` then I guess that is the reason why we need both `loc1` and `loc2`.

Reply:
It's because in Regex and Comby I use line_col_of_charpos, so I just need the starting charpos and the matched string.
But this line_col_of_charpos has a cost, because it parses the file and compute the line x col for each charpos.
Spacegrep actually already has this information, which is why I reuse it instead of calling line_col_of_charpos there.
Actually Comby can also return a line x col, which would again avoid calls to line_col_of_charpos.